pkg/testutils/mockmaps/lbmap.go

```diff
@@ -41,8 +41,12 @@ func (m *LBMockMap) UpsertService(p *lbmap.UpsertServiceParams) error {
 		}
 		backendsList = append(backendsList, *b)
 	}
-	if p.UseMaglev && len(p.ActiveBackends) != 0 {
-		if err := m.UpsertMaglevLookupTable(p.ID, p.ActiveBackends, p.IPv6); err != nil {
+	backends := p.ActiveBackends
+	if len(p.PreferredBackends) > 0 {
+		backends = p.PreferredBackends
```

Comment:
Isn't this code used only in testing? If yes, does it matter whether this is behind a flag?

Reply:
oh right, this file is only for testing purpose. So probably Aditi's point is for pkg/maps/lbmap/lbmap.go as well.

I think it's best to keep generic logic for lbmap, but it might not be possible considered that more clustermesh features are added. For example, with non overlapping IP, we might need to pass cluster id/ cluster name to LB/datapath level. 

The changes in this PR is mainly about additional PreferredBackends attribute, I have added a note saying that this map is only used for clustermesh with service affinity annotation. It's not ideal solution, but could be helpful with your concern.

Reply:
Yeah, the comment was meant to be for `pkg/maps/lbmap` changes.

>I think it's best to keep generic logic for lbmap, but it might not be possible considered that more clustermesh features are added.

Why not? 🤔  I would suggest adding a TO-DO item to revisit this along with separating service mesh related logic.